src/Core/Components/DataGrid/Columns/ColumnBase.razor

```diff
@@ -171,4 +170,18 @@
             @PlaceholderTemplate(placeholderContext)
         }
     }
+
+    protected override void OnParametersSet()
+    {
+        base.OnParametersSet();
+        Grid.AddColumn(this, InitialSortDirection, IsDefaultSortColumn);
```

Comment:
I don't think you can add this `AddColumn` to the `OnParametersSet` method.
This method will be triggered several times: each time a parameter is updated.